
# HG changeset patch
# User Zhao Jiazhong <zhaojiazhong-hf@loongson.cn>
# Date 1631000649 0
# Node ID eac40293649628d32911c909219d514bebb8ce63
# Parent  6582d77235867224aaf8621bad8ee167dfb31460
Bug 1729124 - [MIPS64][ARM64] Fix gcc build issue. r=lth

The SpecificRegs struct needs a constructor.

Differential Revision: https://phabricator.services.mozilla.com/D124535

diff --git a/js/src/wasm/WasmBCRegDefs.h b/js/src/wasm/WasmBCRegDefs.h
--- a/js/src/wasm/WasmBCRegDefs.h
+++ b/js/src/wasm/WasmBCRegDefs.h
@@ -374,25 +374,31 @@ struct SpecificRegs {
 };
 #elif defined(JS_CODEGEN_ARM)
 struct SpecificRegs {
   RegI64 abiReturnRegI64;
 
   SpecificRegs() : abiReturnRegI64(ReturnReg64) {}
 };
 #elif defined(JS_CODEGEN_ARM64)
-struct SpecificRegs {};
+struct SpecificRegs {
+  // Required by gcc.
+  SpecificRegs() {}
+};
 #elif defined(JS_CODEGEN_MIPS32)
 struct SpecificRegs {
   RegI64 abiReturnRegI64;
 
   SpecificRegs() : abiReturnRegI64(ReturnReg64) {}
 };
 #elif defined(JS_CODEGEN_MIPS64)
-struct SpecificRegs {};
+struct SpecificRegs {
+  // Required by gcc.
+  SpecificRegs() {}
+};
 #else
 struct SpecificRegs {
 #  ifndef JS_64BIT
   RegI64 abiReturnRegI64;
 #  endif
 
   SpecificRegs() { MOZ_CRASH("BaseCompiler porting interface: SpecificRegs"); }
 };

